src/i9n/fungible.rs

```diff
@@ -191,25 +191,28 @@ impl Runtime {
     pub fn asset_allocations(
         &mut self,
         contract_id: ContractId,
-    ) -> Result<(), Error> {
+    ) -> Result<Reply, Error> {
         match &*self.command(Request::Allocations(contract_id))? {
             Reply::Failure(failure) => Err(Error::Reply(failure.clone())),
-            Reply::Success => {
+            Reply::Allocations(response) => {
                 println!("Asset allocations succeeded");
 
-                Ok(())
+                Ok(reply::Reply::Allocations(response.clone()))
```

Comment:
No need to wrap object into Reply type here and below. And the return type of the fn can be changed as well

Reply:
I don't get what do you mean, if I remove object wrapping like this
```
Ok(response.clone())
```
it gives me this error:
```
    |                    expected enum `reply::Reply`, found struct `BTreeMap`
    |                    help: try using a variant of the expected enum: `reply::Reply::Allocations(response.clone())`
```
I'm sorry if it's a stupid question but I'm still learning rust. moreover what should be the return type if not `Reply`?

Reply:
Ok sorry, the question was stupid. The function should return a BTreeMap object, right?